src/traces/streamtube/attributes.js

```diff
@@ -0,0 +1,151 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var colorAttrs = require('../../components/colorscale/color_attributes');
+var colorscaleAttrs = require('../../components/colorscale/attributes');
+var colorbarAttrs = require('../../components/colorbar/attributes');
+var mesh3dAttrs = require('../mesh3d/attributes');
+var baseAttrs = require('../../plots/attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+var attrs = {
+    x: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: 'Sets the x coordinates of the vector field.'
+    },
+    y: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: 'Sets the y coordinates of the vector field.'
+    },
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: 'Sets the z coordinates of the vector field.'
+    },
+
+    u: {
+        valType: 'data_array',
+        editType: 'calc',
+        description: 'Sets the x components of the vector field.'
+    },
+    v: {
+        valType: 'data_array',
+        editType: 'calc',
+        description: 'Sets the y components of the vector field.'
+    },
+    w: {
+        valType: 'data_array',
+        editType: 'calc',
+        description: 'Sets the z components of the vector field.'
+    },
+
+    startx: {
+        valType: 'data_array',
+        editType: 'calc',
+        description: [
+            'Sets the x components of the starting position of the streamtubes',
+        ].join(' ')
+    },
+    starty: {
+        valType: 'data_array',
+        editType: 'calc',
+        description: [
+            'Sets the y components of the starting position of the streamtubes',
+        ].join(' ')
+    },
+    startz: {
+        valType: 'data_array',
+        editType: 'calc',
+        description: [
+            'Sets the z components of the starting position of the streamtubes',
+        ].join(' ')
+    },
```

Comment:
would this be better as `starts: {x, y, z}`? I can imagine later on when we figure out automatic starts wanting to tweak them with some other attributes that might be nice to collect all in a sub-container.

Reply:
done in https://github.com/plotly/plotly.js/pull/2658/commits/22f0c7e5396518f8ed0d4fbf45257deb4fa3a7d2